### PR TITLE
Make hex_prefix a part of the merkle_partricia_tree

### LINKED ISSUE #26: Move hex_prefix into merkle_patricia_tree.
hex_prefix should be a part of the merkle_patricia_tree project.

### PR DESCRIPTION
**Pull request by vyorkin**
Closes #26.

### CODE REVIEW

File: apps/merkle_patricia_tree/lib/trie/node.ex [hunk 1]
```diff
@@ -48,15 +48,15 @@ defmodule MerklePatriciaTree.Trie.Node do
...
+    [MerklePatriciaTree.HexPrefix.encode({key, true}), value]
   end
 
   defp encode_node_type({:branch, branches}) when length(branches) == 17 do
     branches
   end
 
   defp encode_node_type({:ext, shared_prefix, next_node}) do
-    [HexPrefix.encode({shared_prefix, false}), next_node]
+    [MerklePatriciaTree.HexPrefix.encode({shared_prefix, false}), next_node]
```
Comment (ayrat555) [inline apps/merkle_patricia_tree/lib/trie/node.ex]: Let's use alias

**Comment by vyorkin** [comment]
Finally

File: apps/merkle_patricia_tree/lib/hex_prefix.ex [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    iex> MerklePatriciaTree.HexPrefix.encode({[ 0, 1, 2, 3, 4, 5 ], false})
+    <<0x00, 0x01, 0x23, 0x45>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[ 0, 15, 1, 12, 11, 8 ], true})
+    <<0x20, 0x0f, 0x1c, 0xb8>>
+
+    iex> MerklePatriciaTree.HexPrefix.encode({[ 15, 1, 12, 11, 8 ], true})
+    <<0x3f, 0x1c, 0xb8>>
+  """
+  @spec encode(__MODULE__.t()) :: binary()
```
Comment (ayrat555) [inline apps/merkle_patricia_tree/lib/hex_prefix.ex]: `__MODULE__.t()` can be replaces with t()

Comment (vyorkin) [reply apps/merkle_patricia_tree/lib/hex_prefix.ex]: thx, didn't notice that

**Pull request merged**